dffml/source/csv.py

```diff
@@ -45,6 +78,11 @@ class CSVSource(FileSource, MemorySource):
                         repo_data["features"][key] = ast.literal_eval(value)
                     except (SyntaxError, ValueError):
                         repo_data["features"][key] = value
+                if self.config.key is not None and self.config.key == key:
+                    src_url = value
+                if self.config.key is None:
+                    src_url = str(i)
+            i += 1
```

Comment:
i += 1 needs to be in the loop

Reply:
Oh, i think it has to outside the loop. Because inside the loop we are looking at only one Repo's key, value pairs.

Reply:
i+=1 still happens inside the main loop which runs for all the repos. It gets incremented only after we assign it to a particular repo. If i move it inside the inner loop as well, then it has to be incremented for each key, value pair from the data in a row. So this should be added outside the inner loop but within the main loop.

Reply:
Ah yes my bad!